### PR TITLE
Adding kiam

### PR DESCRIPTION
**Pull request by nemo83**
Added kiam chart.

I used the `hbc-common-k8s` template as the stable/kiam is very buggy.

I refactored a couple of things so to make it more generic and avoid duplication.

* no need to specify `--namespace` with `kubectl` or in the values file. Namespace has to be specified now via the helm template command, e.g. `helm template --name kiam kiam/ --namespace kiam-system`
*  removed some config values not needed anymore, please check https://raw.githubusercontent.com/saksdirect/personalization/master/charts/kiam/values-personalization.yaml

### CODE REVIEW

File: kiam/templates/agent-daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          args:
+            - --iptables
+            - --host-interface=eni+
+            - --host=$(HOST_IP)
+            - --json-log
+            - --port=8181
+            - --cert=/etc/kiam/tls/agent.pem
+            - --key=/etc/kiam/tls/agent-key.pem
+            - --ca=/etc/kiam/tls/ca.pem
+            - --server-address=kiam.server
```
Comment (conorfennell) [inline kiam/templates/agent-daemonset.yaml]: I am a bit unsure how this resolves to the kiam server.
Does this need to be passed in as a parameter like here https://github.com/saksdirect/hbc-common-k8s/blob/62e58613244e2cd1fda1cea4a0b79d7e8799c7bf/charts/kiam/templates/agent-daemonset.yaml#L59

Comment (nemo83) [reply kiam/templates/agent-daemonset.yaml]: good catch. this is a mistake, not sure what I was thinking when I was modifying it.

the name here can be hardcoded as the name, in the `Service` definition, is not parameterised and cannot be changed. fixing it now.

**Comment by nemo83** [comment]
@conorfennell I've fixed the issue. If you approve. I'll merge :-)

**Comment by nemo83** [comment]
rebased master so to have the latest index.

created new charts and updated index.

**Comment by nemo83** [comment]
@amarrella I released the kiam chart as well. :-)

**Pull request merged**